### PR TITLE
validation: improve header validation

### PR DESCRIPTION
**Pull request by howardjohn**
Actually check they are valid values

ReqAuth -- this is a header *name* not a header *value*. Change check
accordingly

### CODE REVIEW

File: pkg/config/validation/validation.go [hunk 1]
```diff
@@ -228,6 +230,25 @@ func ValidateHTTPHeaderName(name string) error {
...
+	}
+	return nil
+}
+
+// ValidateHTTPHeaderNameOrJwtClaimRoute validates a header name, allowing special @request.auth.claims syntax
+func ValidateHTTPHeaderNameOrJwtClaimRoute(name string) error {
+	if name == "" {
+		return fmt.Errorf("header name cannot be empty")
+	}
+	if jwt.ToRoutingClaim(name).Match {
```
Comment (wulianglongrd) [inline pkg/config/validation/validation.go:244]: Should we cut the `@request.auth.claims` prefix and then validate according to the rules of the header?

File: pkg/config/validation/validation.go [hunk 2]
```diff
@@ -228,6 +230,25 @@ func ValidateHTTPHeaderName(name string) error {
 	if name == "" {
 		return fmt.Errorf("header name cannot be empty")
 	}
+	if !validHeaderRegex.MatchString(name) {
+		return fmt.Errorf("header name %s is not a valid header name", name)
```
Comment (hzxuzhonghu) [inline pkg/config/validation/validation.go:234]: This is a breaking change?

File: pkg/config/validation/validation.go [hunk 3]
```diff
@@ -228,6 +230,25 @@ func ValidateHTTPHeaderName(name string) error {
...
+	}
+	return nil
+}
+
+// ValidateHTTPHeaderNameOrJwtClaimRoute validates a header name, allowing special @request.auth.claims syntax
```
Comment (hzxuzhonghu) [inline pkg/config/validation/validation.go:239]: seems we need to update the api docs as well

File: pkg/config/validation/validation.go (hunk 2 above)
Comment (howardjohn) [reply pkg/config/validation/validation.go:234]: If these are set today they break the traffic at runtime with Envoy anyways, so I think this is probably fine? Usually we are ok with breakage for rare cases that are already broken anyways

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by laiminhtrung1997** [comment]
@howardjohn With this PR, I cannot use the regex "*" for allowHeaders, right? 
```
...
    corsPolicy:
      allowOrigins:
      - exact: https://example.com
      allowMethods:
      - GET
      allowHeaders:
      - '*'
      allowCredentials: false
```

**Comment by howardjohn** [comment]
> @howardjohn With this PR, I cannot use the regex "*" for allowHeaders, right?
> 
> ```
> ...
>     corsPolicy:
>       allowOrigins:
>       - exact: https://example.com
>       allowMethods:
>       - GET
>       allowHeaders:
>       - '*'
>       allowCredentials: false
> ```

Thanks!! https://github.com/istio/istio/pull/52625

**Comment by nijave** [comment]
Breaks matching `:authority` header however the `authority` matcher can be used instead https://istio.slack.com/archives/C37A4KAAD/p1736439462968209

**Comment by howardjohn** [comment]
@nijave I have fixed this in https://github.com/istio/istio/pull/54602. thanks

**Pull request merged**